pdns/pdns_recursor.cc

```diff
@@ -121,6 +121,11 @@ static thread_local std::shared_ptr<std::vector<std::unique_ptr<RemoteLogger>>>
 static thread_local uint64_t t_outgoingProtobufServersGeneration;
 #endif /* HAVE_PROTOBUF */
 
+#ifdef HAVE_FSTRM
+static thread_local std::shared_ptr<std::vector<std::unique_ptr<RemoteLoggerInterface>>> t_frameStreamServers{nullptr};
```

Comment:
I'm not sure why you are not storing these as `FrameStreamLogger` instead of `RemoteLoggerInterface`?

Reply:
The `FrameStreamLogger` class is not available if !HAVE_FRAMESTREAM. But I could make it a dummy object in that case if you think that is more clear.

Reply:
Do you mean if `!HAVE_FSTRM`? But in that case this variable doesn't exist at all? Or am missing something?

Reply:
I mean !HAVE_FSTRM indeed. Sorry for the confusion. The actual problem is in `lwrec.c asyncresolve()`, there the type of the `fstrmLoggers` vector needs to be known.

Reply:
Understood, thanks! I usually prefer a dummy object in that case rather than having to cast, yes, mostly because I find it easier to read afterwards, but that's only my personal preference :-)

Reply:
Well, actually I might share that preference, I'll rework this.